code_sync/code_sync.py

```diff
@@ -2,39 +2,186 @@
 
 import argparse
 import os
+from pathlib import Path
 import subprocess
+import yaml
+from typing import Dict, Union
+
 
 cmd_str = 'watchmedo shell-command --recursive --patterns="{local_dir}*" --command="rsync --filter=\':- .gitignore\' ' \
           '--exclude \'*.ipynb\' --exclude \'.git\' --delete-after -rz --port {port} {local_dir} ' \
           '{target}:{remote_dir}" {local_dir}'
 
 epilog_str = '''
-Example for connecting to LeoMed:
-    code_sync --local_dir mylocaldir/ --remote_dir myremotedir/ --target medinfmk --port 2222\n
+EXAMPLE USAGE
+Register a project:
+    code_sync --register <project>
+
+code_sync a registered project:
+    code_sync <project>
+
+List all projects registered to code_sync:
+    code_sync --list
+
+Run code_sync with specific parameters:
+    code_sync --local_dir <mylocaldir/> --remote_dir <myremotedir/> --target <ssh_remote> --port 2222\n
 
 '''
 
+CONFIG_FILE_NAME = '.code_sync'
+
 
 def code_sync(local_dir, remote_dir, target, port=22):
     # clean up slashes
     local_dir = os.path.join(local_dir, '')
     remote_dir = os.path.join(remote_dir, '')
 
-    # subprocess.call()
+    print("Starting code_sync between {} and {}:{} ...".format(local_dir, target, remote_dir))
+    print('(^C to quit)')
     cmd = cmd_str.format(local_dir=local_dir, remote_dir=remote_dir, target=target, port=port)
     subprocess.call(cmd, shell=True)
 
 
+def load_config() -> Dict:
+    """
+    Load the code_sync config file. Create a blank one if no file exists.
+
+    Returns:
+        The config loaded from the file.
+    """
+
+    create_config_if_not_exists()
+
+    config_file_path = Path(Path.home(), CONFIG_FILE_NAME)
+    with open(config_file_path, 'r') as f:
+        config = yaml.safe_load(f)
+    # if config is empty, return an empty dictionary (not None)
+    if config is None:
+        config = {}
+    return config
+
+
+def init_config() -> None:
```

Comment:
This function can be rolled into a one-liner on line 74 (where it is called from)

Reply:
Are you saying you don't like my tiny function that's only called in one place?! 😋  I would rather leave it, in case later on there is a need to `init_config` without the `if_not_exists` check 🤷‍♀️